### ISSUE TITLE
compiling sources extracted from a later version of pbi-tools works without errors/warnings using an earlier version

### ISSUE THREAD

**Issue by TarekSalha** [pbi-tools/pbi-tools#184]
yesterday, we have been investigating an interesting issue.
One pbi developer extracted a report (with live connection) using the latest preview version of pbi-tools.
In our test ci/cd pipeline, we are currently using the release (rc1) version to compile the artifacts, which is an older version.

Now, the developer had bookmarks in his report and due to a recent change of the repo structure in pbi-tools, these were extracted into an own folder in the src instead of being located in the config.json.
However, after merging the PR the src was being picked up by the ci/cd pipe, but it did not crash. Instead, it just compiled the pbix file WITHOUT the bookmarks. This due to the fact, that it just ignored the folder and was searching for the bookmarks inside the config. We then uploaded it into the service and just found out about the issue, as someone was complaining about the non-working buttons.

Suggestion: When I try to open an excel file with an older version of excel, then it will typically give me an error or at least a warning about the compatibility issue. I would expect pbi-tools to behave the very same way. Just ignoring the version mismatch seems dangerous to me.
what do you think?

**Comment by mthierba** [comment]
Agreed. The extraction format is actually versioned within the `.pbixproj,json` file (created/updated during `pbi-tools extract`):
- RC.1 comes with the "0.11" version
- RC.2 introduces "0.12"
Hence, a "0.11" app could emit a warning when processing "0.12" (or newer) sources. That'll be useful and is easy to achieve.
